### PR TITLE
[JN-1383] [member's choice] json syntax highlighting

### PR DESCRIPTION
**Pull request by connorlbark**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

Adds fancy syntax highlighting/code editor to the survey json editor. 

I evaluated a few options. The "simple" approach is to render a textbox and overlap a `<pre>` with syntax highlighting. But with json files >6000 lines long, like we have, this approach took about 1.5 seconds to render any changes in the textbox.

CodeMirror, on the other hand, despite being a relatively heavyweight solution, is much more cleverly optimized and is able to work with very large json files just fine. The downside being that we're no longer working with a regular textbox, but rather a complex series of divs and an emulated cursor.

Because of this, it doesn't seem easy to test and the current tests are failing. I decided to punt this off to work on other member's choice projects. In my admittedly quick evaulation, I think it might be difficult to get the tests working with the new editor - open to ideas here (putting a good amount of effort into the tests or significantly simplifying them).

image

Also, I ran into weird component refresh issues using it in other places in the app. I figured here was the most important place, and if we like it, we can put it elsewhere in the future. We could also use the markdown module in the website editor, as well.

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

- `npm install`
- try it out - make sure it works with no weirdness

### CODE REVIEW

File: ui-admin/src/util/json/JsonEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import CodeMirror from '@uiw/react-codemirror'
+import { json } from '@codemirror/lang-json'
+import { githubLight } from '@uiw/codemirror-theme-github'
```
Comment (connorlbark) [inline ui-admin/src/util/json/JsonEditor.tsx:4]: there are [other themes](https://uiwjs.github.io/react-codemirror/#/theme/home) if anyone has any preferences

**Review by devonbush — APPROVED** [review]
agree that this seems like the right approach -- might as well have a full-fledged editor given how deeply we work with json in the admin tool.  (And as long as we can lazy-load it as you've done)

File: ui-admin/src/util/json/JsonEditor.tsx (hunk 1 above)
Comment (MatthewBemis) [reply ui-admin/src/util/json/JsonEditor.tsx:4]: Any of the light ones look good imo- so what you've chosen works for me!

**Review by MatthewBemis — APPROVED** [review]
🎉 woohoo! works great

**Pull request merged**